File: sdks/rs/src/integrations/otel/trace.rs
#[derive(Clone, Debug, Default)]
pub struct TraceContext {
    pub trace_id: String,
    pub span_id: String,
}

pub fn enrich_trace(trace_id: impl Into<String>, span_id: impl Into<String>) -> TraceContext {
    TraceContext {
        trace_id: trace_id.into(),
        span_id: span_id.into(),
    }
}

File: sdks/go/src/integrations/otel/trace.go
package otel

import (
	"context"

	"github.com/astraive/loza/sdks/go"
	"go.opentelemetry.io/otel/trace"
)

// TraceAttrs extracts trace context identifiers from ctx as loza attrs.
func TraceAttrs(ctx context.Context) []loza.Attr {
	if ctx == nil {
		return nil
	}
	span := trace.SpanFromContext(ctx)
	sc := span.SpanContext()
	if !sc.IsValid() {
		return nil
	}
	return []loza.Attr{
		loza.TraceID(sc.TraceID().String()),
		loza.SpanID(sc.SpanID().String()),
	}
}

// EnrichTrace extracts and appends trace attrs to an active LOZA event in ctx.
func EnrichTrace(ctx context.Context) {
	attrs := TraceAttrs(ctx)
	if len(attrs) == 0 {
		return
	}
	loza.Enrich(ctx, attrs...)
}
